packages/protocol/contracts/layer1/hekla/HeklaTaikoToken.sol

```diff
@@ -0,0 +1,145 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+import "@openzeppelin/contracts-upgradeable/token/ERC20/ERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/extensions/ERC20SnapshotUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/extensions/ERC20VotesUpgradeable.sol";
+import "../../shared/common/EssentialContract.sol";
+
+/// @title HeklaTaikoToken
+/// @notice The HeklaTaikoToken (TTKOh), in the protocol is used for prover collateral
```

Comment:
how about we also use TKO as the symbol name?

Reply:
We can just add a function symbol() and return "TAIKO"

Reply:
Add this:

/// @dev Due to historical reasons, the Taiko Token on Hekla has a different storage layout compared to the mainnet token contract. Therefore, we need to maintain this file.

Reply:
Added

Reply:
> Add this:
> 
> /// @dev Due to historical reasons, the Taiko Token on Hekla has a different storage layout compared to the mainnet token contract. Therefore, we need to maintain this file.

Done